File: src/app/models/address.ts
import { baseModel } from './baseModel';
import { objectState } from './objectState';
import { Phone } from './phone';

export class Address extends baseModel {
    id: number;
    addressType: string;
    streetAddress1: string;
    streetAddress2: string;
    city: string;
    state: string;
    zipCode: string;
    customerId: number;

    phoneNumbers: Phone[];

    constructor(){
        super()
        this.id = 0;
        this.addressType = 'Other';
        this.streetAddress1 = '';
        this.streetAddress2 = '';
        this.city = '';
        this.state = '';
        this.zipCode = '';
        this.customerId = 0;
        this.phoneNumbers = [
            new Phone()
        ];
    }
}

File: api/Customers.WebApi/Models/Address.cs
using Customers.WebApi.Models;

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CustomerDatabase.Shared
{
    public class Address : BaseModel
    {
        public int Id { get; set; }
		public string AddressType { get; set; }
		public string StreetAddress1 { get; set; }
        public string StreetAddress2 { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string ZipCode { get; set; }

        public int CustomerId { get; set; }

        public virtual Customer? Customer { get; set; }
        public virtual List<PhoneNumber> PhoneNumbers { get; set; }
    }
}
